### PR TITLE
MINOR: Fix DefaultRecordBatch and DefaultRecord protocol documentation.

### PR DESCRIPTION
**Pull request by Veryfirefly**
The documentation for the DefaultRecordBatch protocol does not list recordsCount, which can be ambiguous for someone looking at the RecordBatch protocol through the official documentation or code comments.

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/common/record/DefaultRecordBatch.java [hunk 1]
```diff
@@ -158,7 +159,7 @@ public void ensureValid() {
 
     /**
      * Gets the base timestamp of the batch which is used to calculate the record timestamps from the deltas.
-     * 
+     *
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/common/record/DefaultRecordBatch.java]: could you please remove this unrelated change

**Review by chia7712 — COMMENTED** [review]
@Veryfirefly thanks for your contribution. overall LGTM. will merge it after the comment is addressed

File: clients/src/main/java/org/apache/kafka/common/record/DefaultRecordBatch.java (hunk 1 above)
Comment (Veryfirefly) [reply clients/src/main/java/org/apache/kafka/common/record/DefaultRecordBatch.java]: Sorry for the delay as I was on vacation. I have deleted the unrelated changes.

**Review by chia7712 — APPROVED** [review]
LGTM

**Comment by chia7712** [comment]
@Veryfirefly thanks for your contributions

**Comment by Veryfirefly** [comment]
> @Veryfirefly thanks for your contributions

Thanks, have a nice day
